azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -86,4 +97,26 @@ protected void configure() {
   LocalStorage createLocalStorage(StorageConfig config) {
     return new LocalStorage(new File(config.getBaseDirectoryPath()));
   }
+
+  // todo kunkun-tang: the below method should moved out to azkaban-db module eventually.
+  // Today azkaban-db can not rely on Props, so we can not do it.
+  private static AzkabanDataSource getDataSource(Props props) {
+    String databaseType = props.getString("database.type");
+
+    // todo kunkun-tang: temperaroy workaround to let service provider test work.
+    if(databaseType.equals("h2")) {
```

Comment:
Do we need to specify hard coded settings for h2 and mysql? Are other database types not supported ?